R/sampling_bias_manual.R

```diff
@@ -68,67 +68,89 @@ sampling_bias_manual <- function(occurrences_sf, bias_weights) {
   ### Start checks
   # 1. check input classes
   if (!("sf" %in% class(occurrences_sf))) {
-    cli::cli_abort(c(
-      "{.var occurrences_sf} must be an sf object.",
-      "x" = "You've supplied a {.cls {class(occurrences_sf)}} object."
-    ))
+    cli::cli_abort(
+      c(
+        "{.var occurrences_sf} must be an sf object.",
+        "x" = "You've supplied a {.cls {class(occurrences_sf)}} object."
+      ),
+      class = "gcube_error_class_occurrences_sf"
+    )
   }
   if (!"sf" %in% class(bias_weights)) {
-    cli::cli_abort(c(
-      "{.var bias_weights} must be an sf object.",
-      "x" = "You've supplied a {.cls {class(bias_weights)}} object."
-    ))
+    cli::cli_abort(
+      c(
+        "{.var bias_weights} must be an sf object.",
+        "x" = "You've supplied a {.cls {class(bias_weights)}} object."
+      ),
+      class = "gcube_error_class_bias_weights"
+    )
   }
 
   # 2. Other checks
   # Check if bias_weights has a column named bias_weight
   if (!("bias_weight" %in% names(bias_weights))) {
-    cli::cli_abort(c(
-      "{.var bias_weights} must have a column named `bias_weight`.",
-      "x" = "You've supplied a grid {.var bias_weights} that has column names
-      {names(bias_weights)}."
-    ))
+    cli::cli_abort(
+      c(
+        "{.var bias_weights} must have a column named `bias_weight`.",
+        "x" = "You've supplied a grid {.var bias_weights} that has column names
+        {names(bias_weights)}."
+      ),
+      class = "gcube_error_column_names"
+    )
   }
 
   # Check if crs is the same
   if (sf::st_crs(occurrences_sf) != sf::st_crs(bias_weights)) {
-    cli::cli_abort("sf::st_crs(occurrences_sf) == sf::st_crs(bias_weights) is not TRUE")
+    cli::cli_abort(
+      "sf::st_crs(observations) == sf::st_crs(grid) is not TRUE"
```

Comment:
This isn't a very clear error message to me, what does `st_crs(grid) == TRUE` actually mean? 

Something like occurrences_sf and bias_weights have a different crs, occurrences_sf has crs: ... and bias_weights has crs : ... but they need to be the same. But maybe you can word it better than I can.